processing/src/main/java/io/druid/query/groupby/epinephelinae/SpillingGrouper.java

```diff
@@ -75,24 +77,42 @@ public SpillingGrouper(
       final int bufferGrouperInitialBuckets,
       final LimitedTemporaryStorage temporaryStorage,
       final ObjectMapper spillMapper,
-      final boolean spillingAllowed
+      final boolean spillingAllowed,
+      final int limit,
+      final boolean sortHasNonGroupingFields
   )
   {
     this.keySerde = keySerdeFactory.factorize();
-    this.keyObjComparator = keySerdeFactory.objectComparator();
-    this.grouper = new BufferGrouper<>(
-        buffer,
-        keySerde,
-        columnSelectorFactory,
-        aggregatorFactories,
-        bufferGrouperMaxSize,
-        bufferGrouperMaxLoadFactor,
-        bufferGrouperInitialBuckets
-    );
+    this.keyObjComparator = keySerdeFactory.objectComparator(false);
+    this.defaultOrderKeyObjComparator = keySerdeFactory.objectComparator(true);
+    if (limit > -1) {
```

Comment:
if it is, let us change things so that limit should be a positive integer... limit <= 0 makes no sense